File: app/src/main/java/com/wkz/viewpager/transformer/StackTransformer.java
package com.wkz.viewpager.transformer;

import android.view.View;

import com.nineoldandroids.view.ViewHelper;

/**
 * @desc 堆叠的转换效果
 * @date 2017-06-30
 */
public class StackTransformer extends AbstractBaseTransformer {
    @Override
    protected void onTransform(View page, float position) {
        ViewHelper.setTranslationX(page, position < 0 ? 0F : -page.getWidth() * position);
    }
}

File: view-pager-transformer/src/main/java/com/fphoenixcorneae/transformer/StackTransformer.kt
package com.fphoenixcorneae.transformer

import android.view.View

/**
 * @desc 堆叠的转换效果
 * @date 2017-06-30
 */
class StackTransformer : AbstractBaseTransformer() {

    override fun onTransform(page: View, position: Float) {
        page.translationX = if (position < 0) {
            0f
        } else {
            -page.width * position
        }
    }
}